app/models/report.rb

```diff
@@ -14,4 +23,48 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def execute_save_procedure
+    ActiveRecord::Base.transaction do
+      save!
+      create_mentions
+    end
+    true
+  rescue ActiveRecord::RecordInvalid
+    false
+  end
+
+  def create_mentions
+    collect_mentioned_report_ids.each do |mentioned_report_id|
+      mentioning_relationships.create!(mentioned_id: mentioned_report_id)
+    end
+  end
+
+  def collect_mentioned_report_ids
+    content.scan(%r{http://localhost:3000/reports/(\d+)}).flatten.map(&:to_i)
+  end
+
+  def execute_update_procedure(report_params)
+    ActiveRecord::Base.transaction do
+      update!(report_params)
+      update_mentions
+    end
+    true
+  rescue ActiveRecord::RecordInvalid
+    false
+  end
+
+  def update_mentions
+    old_mentioned_report_ids = mentioning_relationships.map(&:mentioned_id)
```

Comment:
```suggestion
    old_mentioned_report_ids = mentioning_report_ids
```

でも同じデータを取ってこれそうです（となると変数に入れる意味が薄くなりそう）